src/main/java/com/gentics/elasticsearch/client/AbstractElasticsearchClient.java

```diff
@@ -26,20 +38,87 @@ public AbstractElasticsearchClient(String scheme, String hostname, int port) {
 	 * @param parser
 	 *            Parser function for input strings
 	 */
+	@Override
 	public void setConverterFunction(Function<String, T> parser) {
 		this.parser = parser;
 	}
 
+	@Override
 	public int getPort() {
 		return port;
 	}
 
+	@Override
 	public String getScheme() {
 		return scheme;
 	}
 
+	@Override
 	public String getHostname() {
 		return hostname;
 	}
 
+	@Override
+	public void setLogin(String username, String password) {
+		Objects.requireNonNull(username, "Username must not be null");
+		Objects.requireNonNull(password, "Password must not be null");
+		this.username = username;
+		this.password = password;
```

Comment:
How would I clear the login data again?
I'd propose a clearLogin() function for this case

Reply:
You can't the client is immutable. You would need to create a new one using the builder.